sacc/sacc.py

```diff
@@ -595,16 +595,31 @@ def get_mean(self, data_type=None, tracers=None, **select):
         indices = self.indices(data_type=data_type, tracers=tracers, **select)
         return self.mean[indices]
 
-    def get_data_types(self):
+    def get_data_types(self, tracers=None):
         """
         Get a list of the different data types stored in the Sacc
 
+        Parameters
+        ----------
+        tracers: tuple
+            Select only data types which match this tracer combination.
+            If None (the default) then match any tracer combinations.
+
         Returns
         --------
         data_types: list of strings
             A list of the string data types in the data set
         """
-        return unique_list(d.data_type for d in self.data)
+        data_types = unique_list(d.data_type for d in self.data)
+
+        if tracers is not None:
```

Comment:
This will do multiple loops through the data set, which might take a while for large data like supernovae. You can reduce to a single loop by doing something like

```
data_types = unique_list(d.data_type for d in self.data if (tracers is None or d.tracers == tracers))
```
(might need some extra brackets or something there)

Reply:
Great idea! Done